### PR TITLE
[gl][docs] move documentation to source, autogenerate docs

### PR DESCRIPTION
**Pull request by Simek**
# Why

Fixes ENG-1847

# How

This PR moves the current documentation to the `expo-gl` package source and adds an ability to autogenerate docs for the package.

There were also few small changes for the APISection components, which were required to render the newly extracted data correctly, most notable is that, if there is only one component present in the API, their props will be visible in ToC.

Additionally I have squashed `GLViewProps` and `BaseGLViewProps` types definitions together, and removed redeclaration in web component, because otherwise those types were rendered multiple times on the page.

The changes also have been backported to SDK 46.

# Test Plan

The changes have been tested by running the docs website locally. Tests have been updated, lint do not yield any errors.

# Preview

Screenshot 2022-07-21 at 11 01 20

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 1]
```diff
@@ -156,38 +89,6 @@ Since the WebGL API is quite low-level, it can be helpful to use higher-level gr
...
-- `getSyncParameter()`
-- `getActiveUniformBlockParameter()`
-
-The `pixels` argument of [`texImage2D()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texImage2D) must be `null`, an `ArrayBuffer` with pixel data, or an object of the form `{ localUri }` where `localUri` is the `file://` URI of an image in the device's file system. Thus an `Asset` object could be used once `.downloadAsync()` has been called on it (and completed) to fetch the resource.
-
-For efficiency reasons the current implementations of the methods don't perform type or bounds checking on their arguments. So, passing invalid arguments could cause a native crash. We plan to update the API to perform argument checking in upcoming SDK versions. Currently the priority for error checking is low since engines generally don't rely on the OpenGL API to perform argument checking and, even otherwise, checks performed by the underlying OpenGL ES implementation are often sufficient.
-
 ## Integration with Reanimated worklets
 
 To use this API inside Reanimated worklet you need to pass the GL context ID to the worklet and recreate the GL object like in the example bellow.
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/gl-view.md:94]:
```suggestion
To use this API inside Reanimated worklet, you need to pass the GL context ID to the worklet and recreate the GL object like in the example below.
```

Just a small typo fix and some grammar changes(I know this is unrelated to the changes you pushed 😄)

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 2]
```diff
@@ -239,3 +140,43 @@ Check [Reanimated documentation](https://docs.swmansion.com/react-native-reanima
 ## Remote Debugging & GLView
 
 This API does not function as intended with remote debugging enabled. The React Native debugger runs JavaScript on your computer (not the mobile device itself), and GLView requires synchronous native calls (which are not supported in Chrome).
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/gl-view.md:142]:
```suggestion
This API does not function as intended with remote debugging enabled. The React Native debugger runs JavaScript on your computer, not the mobile device. GLView requires synchronous native calls that are not supported in Chrome.
```

I think we can update this paragraph too with this PR. I do not think we need the text inside the brackets since it is part of the explanation. 

Sorry again about the unrelated change 😅

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 3]
```diff
@@ -239,3 +140,43 @@ Check [Reanimated documentation](https://docs.swmansion.com/react-native-reanima
...
+import { GLView } from 'expo-gl';
+```
+
+<APISection packageName="expo-gl" apiName="GLView" />
+
+## WebGL API
+
+Once the component is mounted and the OpenGL ES context has been created, the `gl` object received through the `onContextCreate` prop becomes the interface to the OpenGL ES context, providing a WebGL API. It resembles a [WebGL2RenderingContext](https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7) in the WebGL 2 spec. However, some older Android devices may not support WebGL2 features. To check whether the device supports WebGL2 it's recommended to use `gl instanceof WebGL2RenderingContext`.
+An additional method `gl.endFrameEXP()` is present which notifies the context that the current frame is ready to be presented. This is similar to a 'swap buffers' API call in other OpenGL platforms.
+
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/gl-view.md:156]:
```suggestion
Once the component is mounted and the OpenGL ES context has been created, the `gl` object received through the `onContextCreate` prop becomes the interface to the OpenGL ES context, providing a WebGL API. It resembles a [WebGL2RenderingContext](https://www.khronos.org/registry/webgl/specs/latest/2.0/#3.7) in the WebGL 2 spec.

Some older Android devices may not support WebGL2 features. To check whether the device supports WebGL2 it's recommended to use `gl instanceof WebGL2RenderingContext`.

An additional method `gl.endFrameEXP()` is present, which notifies the context that the current frame is ready to present. This is similar to a 'swap buffers' API call in other OpenGL platforms.

```

Line breaks between these paragraphs make the text more readable to me since there were three different contexts that were talked about in the same paragraph. I think, if that is what you meant by it(?).

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 4]
```diff
@@ -239,3 +140,43 @@ Check [Reanimated documentation](https://docs.swmansion.com/react-native-reanima
...
+- `compressedTexSubImage3D()`
+- `fenceSync()`
+- `isSync()`
+- `deleteSync()`
+- `clientWaitSync()`
+- `waitSync()`
+- `getSyncParameter()`
+- `getActiveUniformBlockParameter()`
+
+The `pixels` argument of [`texImage2D()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texImage2D) must be `null`, an `ArrayBuffer` with pixel data, or an object of the form `{ localUri }` where `localUri` is the `file://` URI of an image in the device's file system. Thus an `Asset` object could be used once `.downloadAsync()` has been called on it (and completed) to fetch the resource.
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/gl-view.md:180]:
```suggestion
The `pixels` argument of [`texImage2D()`](https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/texImage2D) must be `null`, an `ArrayBuffer` with pixel data, or an object of the form `{ localUri }` where `localUri` is the `file://` URI of an image in the device's file system. Thus, an `Asset` object is used once `.downloadAsync()` has been called on it (and completed) to fetch the resource.
```

File: docs/pages/versions/unversioned/sdk/gl-view.md [hunk 5]
```diff
@@ -239,3 +140,43 @@ Check [Reanimated documentation](https://docs.swmansion.com/react-native-reanima
...
+
+For efficiency reasons the current implementations of the methods don't perform type or bounds checking on their arguments. So, passing invalid arguments could cause a native crash. We plan to update the API to perform argument checking in upcoming SDK versions. Currently the priority for error checking is low since engines generally don't rely on the OpenGL API to perform argument checking and, even otherwise, checks performed by the underlying OpenGL ES implementation are often sufficient.
```
Comment (amandeepmittal) [inline docs/pages/versions/unversioned/sdk/gl-view.md:182]:
```suggestion
For efficiency reasons, the current implementations of the methods don't perform type or bounds checking on their arguments. So, passing invalid arguments may cause a native crash. There are plans to update the API to perform argument checking in upcoming SDK versions. 

Currently, the priority for error checking is low since engines generally don't rely on the OpenGL API to perform argument checking; otherwise, checks performed by the underlying OpenGL ES implementation are often sufficient.
```

**Review by amandeepmittal — COMMENTED** [review]
I just suggested some copy-related changes. LGTM overall 👍

**Review by amandeepmittal — APPROVED** [review]

File: packages/expo-gl/CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,9 @@
 
 ### 💡 Others
 
+- Merge `GLViewProps` and `BaseGLViewProps` into one type - `GLViewProps`. ([#18322](https://github.com/expo/expo/pull/18322) by [@Simek](https://github.com/Simek))
+- Rename web specific prop types from `GLViewProps` to `GLViewWebProps`. ([#18322](https://github.com/expo/expo/pull/18322) by [@Simek](https://github.com/Simek))
```
Comment (tsapeta) [inline packages/expo-gl/CHANGELOG.md:14]:
```suggestion
- Rename Web-specific prop types from `GLViewProps` to `GLViewWebProps`. ([#18322](https://github.com/expo/expo/pull/18322) by [@Simek](https://github.com/Simek))
```

**Review by tsapeta — APPROVED** [review]
